File: src/main/java/com/example/coursework/newAnalysis/LoanConditions.java
package com.example.coursework.newAnalysis;

import enums.AdditionalSupport;
import enums.Postponement;
import enums.RepaymentPeriod;
import enums.TypeOfRepayment;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class LoanConditions {
    private int costOfHousing;
    private double PV;

    private TypeOfRepayment typeOfRepayment;
    private Postponement postponement;

    private double minLoanAmount;
    private double maxLoanAmount;
    private double loanAmount;
    private int term;

    private RepaymentPeriod repaymentPeriod;
    private Guarantee guarantee;
    private AdditionalSupport additionalSupport;

    private boolean isRealEstateInsurance;
    private boolean isLifeInsurance;
    private boolean isDoubleBid;

    private double basicPercentage;
    private double percentageIncludingInsurance;
    private double maximumPayment;
    private double averageEPOnRequest;
    private double EPIncludingInsurance;
}

File: src/main/resources/static/js/client/entities/LoanConditions.js
import {Guarantee} from "./Guarantee.js";

export class LoanConditions {
    costOfHousing;
    PV;

    typeOfRepayment;
    postponement;

    minLoanAmount;
    maxLoanAmount;
    loanAmount;
    term;

    repaymentPeriod;
    guarantee;
    additionalSupport;

    isRealEstateInsurance;
    isLifeInsurance;
    isDoubleBid;

    basicPercentage;
    percentageIncludingInsurance;
    maximumPayment;
    averageEPOnRequest;
    EPIncludingInsurance;

    constructor(
        costOfHousing, PV, typeOfRepayment, postponement, minLoanAmount, maxLoanAmount, loanAmount, term,
        repaymentPeriod, guarantee, additionalSupport, isRealEstateInsurance, isLifeInsurance, isDoubleBid,
        basicPercentage, percentageIncludingInsurance, maximumPayment, averageEPOnRequest, EPIncludingInsurance) {
        this.costOfHousing = costOfHousing || 0;
        this.PV = PV || 0;
        this.typeOfRepayment = typeOfRepayment || null;
        this.postponement = postponement || null;
        this.minLoanAmount = minLoanAmount || 0;
        this.maxLoanAmount = maxLoanAmount || 0;
        this.loanAmount = loanAmount || 0;
        this.term = term || 0;
        this.repaymentPeriod = repaymentPeriod || null;
        this.guarantee = guarantee || new Guarantee();
        this.additionalSupport = additionalSupport || null;
        this.isRealEstateInsurance = isRealEstateInsurance || false;
        this.isLifeInsurance = isLifeInsurance || false;
        this.isDoubleBid = isDoubleBid || false;
        this.basicPercentage = basicPercentage || 0;
        this.percentageIncludingInsurance = percentageIncludingInsurance || 0;
        this.maximumPayment = maximumPayment || 0;
        this.averageEPOnRequest = averageEPOnRequest || 0;
        this.EPIncludingInsurance = EPIncludingInsurance || 0;
    }
}
